### ISSUE TITLE
Finetung with Trajectory Condiiton

### ISSUE THREAD

**Issue by OrangeSodahub** [alibaba/Tora#28]
Hi, thanks for your great contributions! I have a question, when you initialized from pretrained weights, and you add the addition trajectory condiiton (as embedding added to time embedding), then through the AdaLN. I want to make sure that you trained **all** the parameters of the model? Since you changed the embedding input to the AdaLN. Have you tried other finetuned settings, such as freeze part of the parameters.

In my case, I freeze the pretrained weigths and only train the trajectory condiiton weights, it totally failed. And training all parameters leads to extremely low efficiency.

Any suggestions would be appreciate!

**Comment by zhenghao977** [comment]
There are two approaches to training the base model without updating all its parameters.

First, you can focus on training only the Tora module as an adapter. This method allows the model to train much faster. While the generated results might look more visually stable, the control over motion in the output could be somewhat limited.

Second, you can adjust all the parameters within the AdaLNMixin module. This approach balances both the visual quality of the outputs and the level of motion control you can achieve.

**Comment by OrangeSodahub** [comment]
@zhenghao977 Thanks for your useful suggestions. I have some points wanted to clear:

Lets take the diffusers versions as the example, simply you add two components to CogvideoX block: 1) trajectory extractor: it's trained separately; 2) the fuser module: where I note you append the zero module at the output layer. The trained fuser part is:
https://github.com/alibaba/Tora/blob/31e57bb2fd6f4507e0da403ff869389a55345f9f/diffusers-version/tora/transformer_3d.py#L140-L145

If I am right, then the first choice as you suggested is to train only the ''fuser'' module; the second choice is extend the training part to the whole normalzation layer. If something was wrong, you can point out, thanks!

**Comment by OrangeSodahub** [comment]
@zhenghao977 I've tried training only the fuser module in diffusers' implementation. But it did not work at all.

I was wondering, which parts I should unfreeze to train in diffusers version, could you give some advice? Thanks!

**Comment by OrangeSodahub** [comment]
I just found that I am possibly right, the only trainable modules are the fuser and trajectory extractor:
https://github.com/alibaba/Tora/blob/31e57bb2fd6f4507e0da403ff869389a55345f9f/sat/configs/tora/model/cogvideox_5b_tora.yaml#L6-L9

But, I tried another way, I directly encode the trajectory into embeddings, and input to the fuser along with image latents, however it seems not work.

**Comment by zhenghao977** [comment]
@OrangeSodahub  Consider adjusting all relevant hyperparameters in the AdaLNMixin module and reducing the learning rate in your experiments.

**Comment by OrangeSodahub** [comment]
@zhenghao977 Thanks for your suggestions! But as I described, I followed your **diffusers** version. And set only `fuser` trainable, and it failed.

**Comment by massyzs** [comment]
Hi, may I ask if you faced a problem when loading TORA parameters? Something like "could not find the metadata file ckpts/tora/i2v/mp_rank_00_model_states.pt/latest "

**Comment by antragoudaras** [comment]
> There are two approaches to training the base model without updating all its parameters.
> 
> First, you can focus on training only the Tora module as an adapter. This method allows the model to train much faster. While the generated results might look more visually stable, the control over motion in the output could be somewhat limited.
> 
> Second, you can adjust all the parameters within the AdaLNMixin module. This approach balances both the visual quality of the outputs and the level of motion control you can achieve.

@zhenghao977 Do you mean in the first approach that you tried a LoRA like adapter, and followed the [configs of CogVideoX](https://github.com/THUDM/CogVideo/blob/main/sat/configs/cogvideox_5b_lora.yaml)for it? Or did you tried something different?
